infra/baseServer.go

```diff
@@ -37,17 +27,16 @@ type BaseServer[T IAgent[T]] struct {
 
 func (server *BaseServer[T]) HandleStartOfTurn(iter, round int) {
 	fmt.Printf("Iteration %d, Round %d starting...\n", iter, round)
-	server.beginAgentListeningSession()
 
 }
 
-func (serv *BaseServer[T]) waitForMessagingToEnd() {
+func (serv *BaseServer[T]) endAgentListeningSession() {
+
 	timeoutChannel := time.After(serv.maxMessagingDuration)
 
 agentMessaging:
 	for {
 		select {
-
 		case <-timeoutChannel:
```

Comment:
this seems to be achievable with the following:

```suggestion
		case <-time.After(*some time*): 
                // some logic
```

Reply:
I.e., avoids the spinner entirely